cpu/native/periph/random.c

```diff
@@ -40,33 +72,89 @@ int random_read(char *buf, unsigned int num)
         return 0;
     }
 
-    DEBUGF("random_read: writing %ui bytes\n", num);
-    for (unsigned i = 0; i < num; i++) {
-        _native_syscall_enter();
-        buf[i] = (char)real_random();
-        _native_syscall_leave();
+    DEBUG("random_read: writing %u bytes\n", num);
+    switch (_native_rng_mode) {
+        case 0:
+            num = _native_rng_read_hq(buf, num);
+            break;
+        case 1:
+            num = _native_rng_read_det(buf, num);
+        default:
+            DEBUG("random_read: _native_rng_mode is in invalid state %i\n",
+                   _native_rng_mode);
```

Comment:
It is not possible for the variable to take a different value unless someone modified the implementation, so it does not need to be communicated in my opinion.